### PR TITLE
test(render_detail): cover untested `_build_event_details` branches (#860)

### LINKED ISSUE #860: [aw][test audit] render_detail: untested branches in `_build_event_details` — SESSION_SHUTDOWN falsy type and TOOL_EXECUTION mod
[Content truncated due to length]
## Summary

Two code branches in `render_detail._build_event_details` are never exercised by the current test suite, so regressions there can't be caught by CI.

---

## Gap 1 — `SESSION_SHUTDOWN` with `shutdownType=None`

**Location:** `src/copilot_usage/render_detail.py:156`

```python
case EventType.SESSION_SHUTDOWN:
    if (data := _safe_event_data(ev, ev.as_session_shutdown)) is None:
        return ""
    return f"type={data.shutdownType}" if data.shutdownType else ""
```

`SessionShutdownData.shutdownType` is typed `str | None`, so the field is legitimately absent in some real events. When the field is `None` or `""`, the function returns `""`.

`TestBuildEventDetails.test_session_shutdown_details` only exercises the truthy case (`shutdownType="routine"`). The **else branch** (`shutdownType is None` → `""`) has zero test coverage.

**Expected test:** An `EVENT_TYPE.SESSION_SHUTDOWN` event with a `SessionShutdownData` whose `shutdownType` is `None` (or omitted from the raw `data` dict) should make `_build_event_details` return `""`.

---

## Gap 2 — `TOOL_EXECUTION_COMPLETE` with non-`None` `model` field

**Location:** `src/copilot_usage/render_detail.py:149–150`

```python
if data.model:
    parts.append(f"model={data.model}")
```

`ToolExecutionData.model` is optional. When present it should appear in the detail string as `model=<name>`. All three existing `TestBuildEventDetails` tests for `TOOL_EXECUTION_COMPLETE` omit the `model` field from the raw dict, so the truthy branch is never reached.

**Expected test:** A `TOOL_EXECUTION_COMPLETE` event whose raw data includes `"model": "claude-sonnet-4"` should produce a details string containing `"model=claude-sonnet-4"`.

---

## Regression risk

Both are **display-path branches** — a silent regression would cause incorrect or missing information in `copilot-usage session <id>` detail output without failing any test. The branches are simple enough that a single parametrized test covers each case.

> Generated by [Test Suite Analysis](https://github.com/microsasa/cli-tools/actions/runs/24139465807/agentic_workflow) · ● 8.6M · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsasa%2Fcli-tools%2Ftest-analysis%22&type=issues)

### PR DESCRIPTION
**Pull request by microsasa**
Closes #860

Adds two tests in `TestBuildEventDetailsUntestedBranches` covering the two branches in `_build_event_details` that had zero test coverage:

1. **`SESSION_SHUTDOWN` with `shutdownType=None`** — verifies the else branch returns `""` when `shutdownType` is falsy (omitted from raw data).
2. **`TOOL_EXECUTION_COMPLETE` with `model` field** — verifies the truthy branch includes `model=claude-sonnet-4` in the detail string.

All existing tests continue to pass. `make check` passes cleanly (lint, typecheck, security, 99% coverage).

> Generated by [Issue Implementer](https://github.com/microsasa/cli-tools/actions/runs/24148508672/agentic_workflow) · ● 5.5M · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+%22gh-aw-workflow-id%3A+issue-implementer%22&type=pullrequests)

### CODE REVIEW

File: tests/copilot_usage/test_render_detail.py [hunk 1]
```diff
@@ -608,3 +608,27 @@ def test_multi_model_shutdown_via_full_render(self) -> None:
...
+
+
+# ---------------------------------------------------------------------------
+# Issue #860 — untested branch in _build_event_details
+# ---------------------------------------------------------------------------
+
+
+class TestBuildEventDetailsUntestedBranches:
+    """Cover the remaining _build_event_details branch unique to this module."""
+
```
[the remark this answers is not in the document]
Comment (microsasa) [reply tests/copilot_usage/test_render_detail.py]: Good catch. Added the missing `test_session_shutdown_falsy_shutdown_type` test that directly verifies `_build_event_details` returns `""` when `shutdownType` is falsy (empty dict data). The redundant `TOOL_EXECUTION_COMPLETE` test was removed per the other review comment since it's already covered in `test_report.py`.

> Generated by [Review Responder](https://github.com/microsasa/cli-tools/actions/runs/24149657305/agentic_workflow) · ● 4.2M

**Comment by microsasa** [comment]
Closing — this PR (a 39-line test addition) went through 6 rounds of Copilot code review (9 threads). The agent addressed 7/9, with 2 unresolved.

This one is particularly egregious: **Copilot contradicted itself across rounds**, making the agent ping-pong between adding and removing tests:
- Round 2: "SESSION_SHUTDOWN test is redundant with test_report.py, remove it"
- Round 3: "TOOL_EXECUTION test is also redundant, delete the whole block"
- Round 4: "Add the SESSION_SHUTDOWN test back, it's missing"
- Round 5: "PR says two tests but only has one, add the missing one"
- Round 6: "That TOOL_EXECUTION test is redundant, delete it" (same comment as round 3)

The agent dutifully followed each contradictory instruction, burning 6 responder dispatches on a trivial test PR. The remaining 2 unresolved threads are Copilot saying (again) that the tests are redundant — the same feedback it gave in round 2 that it then reversed in round 4.

PR is 6 days stale with merge conflicts. Untagging #860 for a fresh attempt.
